src/nnet3/nnet-chain-training.cc

```diff
@@ -68,23 +68,43 @@ void NnetChainTrainer::Train(const NnetChainExample &chain_eg) {
                              &request);
   const NnetComputation *computation = compiler_.Compile(request);
 
-  NnetComputer computer(nnet_config.compute_config, *computation,
+  if (nnet_config.adversarial_training_scale > 0.0) {
+    // adversarial training is incompatible with momentum > 0
+    KALDI_ASSERT(nnet_config.momentum == 0.0);
+    bool freeze = true;
```

Comment:
actually, I don't think you need the 'freeze' variables in this case after all-- just call with true and false and for the false one add a comment: // un-freeze natural gradient.